src/Playwright/Helpers/Paths.cs

```diff
@@ -34,18 +35,55 @@ internal static class Paths
         internal static string GetExecutablePath()
         {
             var assembly = typeof(Playwright).Assembly;
-            string driversPath = new FileInfo(assembly.Location).Directory.FullName;
+            string driversPath, executableFile;
 
-            string executableFile = GetPath(driversPath);
+            // in .NET 6, the notion of single-assembly apps means, we won't have a location
+            // and this can throw an exception - let's guard against it
+            if (!string.IsNullOrEmpty(assembly.Location))
+            {
+                driversPath = new FileInfo(assembly.Location).Directory.FullName;
+
+                executableFile = GetPath(driversPath);
+                if (File.Exists(executableFile))
+                {
+                    return executableFile;
+                }
+            }
+
+            // fallback to the NuGet repository
+            driversPath = GetNugetPackagePath(assembly.GetName().Version.ToString(3));
+            executableFile = GetPath(driversPath, "Drivers");
```

Comment:
Can we use .playwright to align? This Drivers vs .playwright thing confuses me all the time!

Reply:
Yes. Done.